sway/input/seatop_down.c

```diff
@@ -2,21 +2,100 @@
 #include <float.h>
 #include <wlr/types/wlr_cursor.h>
 #include <wlr/types/wlr_tablet_v2.h>
+#include <wlr/types/wlr_touch.h>
 #include "sway/input/cursor.h"
 #include "sway/input/seat.h"
 #include "sway/tree/view.h"
 #include "sway/desktop/transaction.h"
 #include "log.h"
 
+struct seatop_touch_point_event {
+	double ref_lx, ref_ly;         // touch's x/y at start of op
+	double ref_con_lx, ref_con_ly; // container's x/y at start of op
+	int32_t touch_id;
+	struct wl_list link;
+};
+
 struct seatop_down_event {
 	struct sway_container *con;
 	struct sway_seat *seat;
 	struct wl_listener surface_destroy;
 	struct wlr_surface *surface;
 	double ref_lx, ref_ly;         // cursor's x/y at start of op
 	double ref_con_lx, ref_con_ly; // container's x/y at start of op
+	struct wl_list point_events;   // seatop_touch_point_event::link
 };
 
+static void handle_touch_motion(struct sway_seat *seat,
+		struct wlr_event_touch_motion *event) {
+	struct seatop_down_event *e = seat->seatop_data;
+
+	struct seatop_touch_point_event *point_event;
+	wl_list_for_each(point_event, &e->point_events, link) {
+		if (point_event->touch_id == event->touch_id) {
+			break;
```

Comment:
What if the point wasn't found in the list? I'd prefer adding ` bool found` variable with `sway_assert(found)` if it's supposed to always be there.

Reply:
okay I added this check